firewood/src/merkle/node.rs

```diff
@@ -315,7 +315,7 @@ impl Node {
     }
 }
 
-const ENCODED_MAX_LEN: usize = TRIE_HASH_LEN;
+const ENCODED_MAX_LEN: usize = TRIE_HASH_LEN - 1;
```

Comment:
Hmm, `is_encoded_longer_than_hash_len` is a bad naming, do you mind changed it to `is_encoded_longer_than_or_equal_to_hash_len` or keep it as it is an update `node::is_encoded_longer_than_hash_len()` to really reflect the naming. Thanks!

Reply:
I created an issue
https://github.com/ava-labs/firewood/issues/508